src/PositionVelocityTime.jl

```diff
@@ -21,30 +23,127 @@ module PositionVelocityTime
         carrier_phase::CP = 0
     end
 
+    """
+    PVTSolution, with used sats and el and az.
+    """
+    @with_kw mutable struct PVTSolution
+        pos::ECEF
+        receiver_time_correction::Float64
+        GDOP::Union{Nothing, Float64}
+        PDOP::Union{Nothing, Float64} 
+        VDOP::Union{Nothing, Float64} 
+        HDOP::Union{Nothing, Float64} 
+        TDOP::Union{Nothing, Float64}
+        #following is added for satllite observation
+        num_used_sats::Int64
+        used_sats::Union{Nothing,AbstractVector{Int64}} 
```

Comment:
`num_used_sats` will be the same as `length(used_sats)`, if `used_sats` is a vector, is that correct?
Instead of having a redundant information here, you can create a helper function:
```
get_num_used_sats(pvt_solution::PVTSolution) = length(used_sats)
```
and instead of using the union of `Nothing` and `AbstractVector`, I would use just the `AbstractVector` and initialize it with an empty vector.